client/resolvers.go

```diff
@@ -15,6 +15,31 @@ func ResolveProject(_ context.Context, meta schema.ClientMeta, r *schema.Resourc
 	return r.Set("project_id", client.ProjectId)
 }
 
+func AddGcpMetadata(ctx context.Context, c schema.ClientMeta, r *schema.Resource) error {
```

Comment:
Most of this change is related to moving the private `resolveLocation` to the bottom.